src/systems.jl

```diff
@@ -114,45 +114,45 @@ function get_mappings(model::SBML.Model)
     inits = Dict(SBML.initial_amounts(model, convert_concentrations = true))
     u0map = Pair[]
     parammap = Pair[]
+    initial_assignment_map = Pair[]
+
     for (k, v) in model.species
+        var = create_symbol(k, model)
         if v.constant == true
-            var = create_param(k; isconstantspecies = true)
             push!(parammap, var => inits[k])
         else
-            var = create_var(k, IV;
-                isbcspecies = has_rule_type(k, model, SBML.RateRule) ||
-                              has_rule_type(k, model, SBML.AssignmentRule) ||
-                              (has_rule_type(k, model, SBML.AlgebraicRule) &&
-                               (all([netstoich(k, r) == 0
-                                     for r in values(model.reactions)]) ||
-                                v.boundary_condition == true)))  # To remove species that are otherwise defined
             push!(u0map, var => inits[k])
         end
     end
+
     for (k, v) in model.parameters
+        var = create_symbol(k, model)
         if v.constant == false &&
            (SBML.seemsdefined(k, model) || is_event_assignment(k, model))
-            var = create_var(k, IV; isbcspecies = true)
             push!(u0map, var => v.value)
         elseif v.constant == true && isnothing(v.value)  # Todo: maybe add this branch also to model.compartments
-            var = create_param(k)
             val = model.initial_assignments[k]
             push!(parammap, var => interpret_as_num(val, model))
         else
-            var = create_param(k)
             push!(parammap, var => v.value)
         end
     end
+
     for (k, v) in model.compartments
+        var = create_symbol(k, model)
         if v.constant == false && SBML.seemsdefined(k, model)
-            var = create_var(k, IV; isbcspecies = true)
             push!(u0map, var => v.size)
         else
-            var = create_param(k)
             push!(parammap, var => v.size)
         end
     end
-    u0map, parammap
+
+    for (k, v) in model.initial_assignments
+        var = create_symbol(k, model)
+        val = model.initial_assignments[k]
+        push!(initial_assignment_map, var => interpret_as_num(val, model))  # Todo
```

Comment:
What's the todo here about?

Reply:
It is already done, but I forgot to delete the comment.🙈